web-services-app/src/test/java/edu/unc/lib/boxc/web/services/rest/DownloadImageControllerIT.java

```diff
@@ -0,0 +1,144 @@
+                        .withStatus(HttpStatus.OK.value())
+                        .withBodyFile("bunny.jpg")
+                        .withHeader("Content-Type", "image/jpeg")));
+
+        when(solrSearchService.getObjectById(any(SimpleIdRequest.class))).thenReturn(record);
+        when(record.getDatastreamObject("original_file")).thenReturn(datastream);
+        when(datastream.getExtent()).thenReturn("1200x1200");
+
+        MvcResult result = mvc.perform(get("/downloadImage/" + pidString + "/800"))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        // Verify content was retrieved
+        MockHttpServletResponse response = result.getResponse();
+
+        assertEquals("image/jpeg", response.getContentType());
+        assertEquals("attachment; filename=image_800.jpg", response.getHeader(CONTENT_DISPOSITION));
+    }
+
+    @Test
+    public void testGetImageAtPixelSizeBiggerThanFull() throws Exception {
+        var pidString = makePid().getId();
+        ContentObjectSolrRecord record = mock(ContentObjectSolrRecord.class);
+        Datastream datastream = mock(Datastream.class);
+
+        stubFor(WireMock.get(urlMatching("/" + pidString + "/full/full/0/default.jpg"))
+                .willReturn(aResponse()
+                        .withStatus(HttpStatus.OK.value())
+                        .withBodyFile("bunny.jpg")
+                        .withHeader("Content-Type", "image/jpeg")));
+
+        when(solrSearchService.getObjectById(any(SimpleIdRequest.class))).thenReturn(record);
+        when(record.getDatastreamObject("original_file")).thenReturn(datastream);
+        when(datastream.getExtent()).thenReturn("1200x1200");
+
+        MvcResult result = mvc.perform(get("/downloadImage/" + pidString + "/2500"))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        // Verify content was retrieved
+        MockHttpServletResponse response = result.getResponse();
+
+        assertEquals("image/jpeg", response.getContentType());
+        assertEquals("attachment; filename=image_full.jpg", response.getHeader(CONTENT_DISPOSITION));
+    }
+
+    @Test
+    public void testFullSizeAccessImageNoPermissions() throws Exception {
+        PID filePid = makePid();
+        ContentObjectSolrRecord record = mock(ContentObjectSolrRecord.class);
+        when(solrSearchService.getObjectById(any(SimpleIdRequest.class))).thenReturn(record);
+        doThrow(new AccessRestrictionException()).when(accessControlService)
+                .assertHasAccess(anyString(), eq(filePid), any(AccessGroupSetImpl.class), eq(viewOriginal));
+
+        MvcResult result = mvc.perform(get("/downloadImage/" + filePid.getId() + "/full"))
+                .andExpect(status().isForbidden())
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        assertEquals("Insufficient permissions", response.getContentAsString());
+    }
+}
```

Comment:
Another case worth checking could be `testGetImageAtPixelSizeBiggerThanFullNoPermission`

Also, what happens if we provide bad input like a non-numerical size, such as "large"? Does it throw an error? Or a blank size? Or a negative size? Some of those probably should throw an error.

What happens if the object doesn't have an original_file datastream? This would happen for any object type other than a File.

Reply:
For the non-numerical size I will throw an illegal argument exception. For the case where there is no original file, how do you feel about a not found exception? @bbpennel